modular_citadel/code/game/turfs/cit_turfs.dm

```diff
@@ -1,24 +1,18 @@
-//Yes, hi. This is the file that handles Citadel's turf modifications.
+/turf/open/floor
+	var/dirtlevel
+	var/candirtify = TRUE
 
-/*
 /turf/open/floor/Entered(atom/obj, atom/oldloc)
 	. = ..()
-	CitDirtify(obj, oldloc)*/
-
-//Baystation-styled tile dirtification.
-/turf/open/floor/proc/CitDirtify(atom/obj, atom/oldloc)
-	if(prob(50))
-		if(has_gravity(src) && !isobserver(obj))
-			var/dirtamount
-			var/obj/effect/decal/cleanable/dirt/dirt = locate(/obj/effect/decal/cleanable/dirt, src)
-			if(!dirt)
-				dirt = new/obj/effect/decal/cleanable/dirt(src)
-				dirt.alpha = 0
-				dirtamount = 0
-			dirtamount = dirt.alpha + 1
-			if(oldloc && istype(oldloc, /turf/open/floor))
-				var/obj/effect/decal/cleanable/dirt/spreadindirt = locate(/obj/effect/decal/cleanable/dirt, oldloc)
-				if(spreadindirt && spreadindirt.alpha)
-					dirtamount += round(spreadindirt.alpha * 0.05)
-			dirt.alpha = min(dirtamount,255)
-	return TRUE
+	if(candirtify && !isobserver(obj))
+		dirtlevel++
+		if(oldloc && istype(oldloc, /turf/open/floor))
+			var/turf/open/floor/F = oldloc
+			dirtlevel += (F.dirtlevel)*0.025
+		if(dirtlevel >= 255)
+			if(!isobserver(obj))
+				var/obj/effect/decal/cleanable/dirt/dirt = locate(/obj/effect/decal/cleanable/dirt, src)
+				if(!dirt)
+					dirt = new/obj/effect/decal/cleanable/dirt/natural(src)
+				dirtlevel = min(dirtlevel, 510)
```

Comment:
This is set in the turf rather than the natural dirt decal because this is the most direct and performant way to handle setting the dirt level. The other two options possible here would be to either call a proc on the natural dirt decal (which introduces more proccall overhead), or to spawn a new natural dirt decal every single time something enters a dirty-able turf (which, dear god why would anyone ever consider that)
The reason why the parent type sets the turf dirtlevel is because quite a few things already use the base cleanable dirt type for dirt generation, so having the base dirt type set the dirtlevel makes dirt spawned forcefully interact naturally with normal dirt, making the forcefully spawned dirt function more or less the same as the natural dirt